progress/meter.go

```diff
@@ -32,24 +33,70 @@ type ProgressMeter struct {
 	dryRun            bool
 }
 
-// NewProgressMeter creates a new ProgressMeter for the number and size of
-// files given.
-func NewProgressMeter(estFiles int, estBytes int64, dryRun bool, logPath string) *ProgressMeter {
-	logger, err := newProgressLogger(logPath)
-	if err != nil {
+type env interface {
+	Get(key string) (val string, ok bool)
+}
+
+type meterOption func(*ProgressMeter)
+
+func DryRun(dryRun bool) meterOption {
+	return func(m *ProgressMeter) {
+		m.dryRun = dryRun
+	}
+}
+
+func WithLogFile(name string) meterOption {
+	printErr := func(err string) {
 		fmt.Fprintf(os.Stderr, "Error creating progress logger: %s\n", err)
 	}
 
-	return &ProgressMeter{
-		logger:         logger,
+	return func(m *ProgressMeter) {
+		if len(name) == 0 {
+			return
+		}
+
+		if !filepath.IsAbs(name) {
+			printErr("GIT_LFS_PROGRESS must be an absolute path")
```

Comment:
Not sure I dig using the `printErr` function like this. What do you think about:

```go
return func(m *ProgressMeter) {
        var err error
        defer func() {
                if err == nil {
                        return
                }

                err = errors.Wrap("Error creating progress logger", err)
                fmt.Fprintf(os.Stderr, "%s", err)
        }()

        if len(name) == 0 {
                return
        }

        if !filepath.IsAbs(name) {
                err = errors.New("GIT_LFS_PROGRESS must be an absolute path")
                return
        }
}
```